### ISSUE TITLE
Sketcher: Add directed horizontal and vertical distance constraints to prevent sketch flipping

### ISSUE THREAD

**Issue by chrisb-github** [FreeCAD/FreeCAD#10708]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

A set of constraints can describe multiple sketches, even if it is fully constrained; this is known as _flipping_.
I propose to add new constraints, "directed horizontal dimension" and "directed vertical dimension".

**directed horizontal dimension**
Applied to a geometric element, the direction is from left to right, starting at the start point of the geometric element going to the endpoint.
A positive value would go from left to right, a negative value would go from right to left.

Applied to different geometric elements, the startpoint would be the element with the lower index, the endpoint that with the higher index.

**directed vertical dimension**
This is equivalent with positive values going up and negative values going down.

### Full version info

```shell
OS: macOS 13.0
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.34137 (Git)
Build type: Release
Branch: master
Hash: 7ca834acc58841dcf2d66d186db818c8f35f359d
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: C/Default (C)
Installed mods: 
  * FC_SU
  * freecad.gears 1.0.0
  * FeedsAndSpeeds 0.5.0
  * dxf-library
  * fcgear 1.0.0
  * fasteners 0.4.59
  * sheetmetal 0.2.62
  * ExplodedAssembly
  * Curves 0.6.8
```

### Subproject(s) affected?

Sketcher

### Anything else?

[Forum discussion](https://forum.freecad.org/viewtopic.php?p=707208#p707208).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by qewer33** [comment]
Instead of new constraints, imo it would be better to make the current horizontal/vertical distance constraints work with +/- prefixes. 

So for example when the value for an horizontal distance constraint is "+15" instead of just "15", the solver will make that constraint from left to right (in the +x direction if you think of the sketch plane as a coordinate system). And right to left if the value is "-15" (-x in the coordinate system). 

Normal values without +/- prefixes would work how they work currently, no changes there.

Pretty much exactly what is proposed here but making it work with the current commands instead of adding new commands.

**Comment by Jolbas** [comment]
I think a better idea would be to add a direction lock option to all angular constraints (horisontal, vertical, angle, parallell and perpendicular and maybe tangential). The lock would weigh in the order of the endpoints in the involved lines so that if one line turns 180° the other line will turn too. I think this lock may even be bound to if it is construction geometry or not.

**Comment by r-nd-m** [comment]
This feature is a must have. After updating to 1.0 I'm seeing a lot of flipped constraints in my migrated models - it's frustrating to a point that  I'm considering abandoning FreeCAD if I have to redo all my broken models every major revision, e.g. my PC I/O plate:
<img width="1492" height="664" alt="Image" src="https://github.com/user-attachments/assets/ed209f16-be47-42cb-b4af-ebb993f3bb50" />
I've read some suggestions on the FreeCAD forum to use the origin point for dimensions but that would make dimensioning of any drawing feature a PITA as you cannot use any relative measurements (which still doesn't prevent the issue in 100% of cases!).
Is there any way to setup a bounty?

**Comment by theo-vt** [comment]
Hi hello, just so that it's on paper, I have started working on this and I hope to have a POC by december

**Comment by FlachyJoe** [comment]
FYI #24242 can be used for workaround

**Comment by Roy-043** [comment]
FYI: the current horizontal and vertical dimensions already have directional properties.
https://wiki.freecad.org/Sketcher_Dimension#Notes

**Comment by theo-vt** [comment]
That's right! I think that what is missing is for all cases where horizontal and vertical lines are used

1) A way to interpret dimensions as X/Y dimensions
2) A way to interpret equal length constraints as X/Y dimensions
3) A way to interpret perpendicular/parallel lines as horizontal/vertical
